bedrock-devnet/devnet/__init__.py

```diff
@@ -33,88 +34,112 @@ def main():
-        deploy_config = read_json(deploy_config_path)
-        deploy_config['l1GenesisBlockTimestamp'] = utc_time
-        temp_deploy_config = pjoin(paths.devnet_dir, 'deploy-config.json')
-        write_json(temp_deploy_config, deploy_config)
+        response = json.loads(res)
+        account = response['result'][0]
 
-        outfile_l1 = paths.genesis_l1_path
-        outfile_l2 = paths.genesis_l2_path
-        outfile_rollup = paths.rollup_config_path
+        fqn = 'scripts/Deploy.s.sol:Deploy'
+        run_command([
+            'forge', 'script', fqn, '--sender', account,
+            '--rpc-url', 'http://127.0.0.1:8545', '--broadcast',
+            '--unlocked'
+        ], env={}, cwd=paths.contracts_bedrock_dir)
 
-        run_command(['go', 'run', 'cmd/main.go', 'genesis', 'devnet', '--deploy-config', temp_deploy_config, '--outfile.l1', outfile_l1, '--outfile.l2', outfile_l2, '--outfile.rollup', outfile_rollup], cwd=paths.op_node_dir)
-        write_json(done_file, {})
+        shutil.copy(paths.l1_deployments_path, paths.addresses_json_path)
 
-    log.info('Bringing up L1.')
-    run_command(['docker-compose', 'up', '-d', 'l1'], cwd=paths.ops_bedrock_dir, env={
-        'PWD': paths.ops_bedrock_dir
-    })
-    wait_up(8545)
-    wait_for_rpc_server('127.0.0.1:8545')
+        log.info('Syncing contracts.')
+        run_command([
+            'forge', 'script', fqn, '--sig', 'sync()',
+            '--rpc-url', 'http://127.0.0.1:8545'
+        ], env={}, cwd=paths.contracts_bedrock_dir)
 
-    log.info('Bringing up L2.')
-    run_command(['docker-compose', 'up', '-d', 'l2'], cwd=paths.ops_bedrock_dir, env={
-        'PWD': paths.ops_bedrock_dir
-    })
-    wait_up(9545)
-    wait_for_rpc_server('127.0.0.1:9545')
+    return internal
+
+
+def devnet_l1_genesis(paths):
+    log.info('Generating L1 genesis state')
+    geth = subprocess.Popen([
+        'geth', '--dev', '--http', '--http.api', 'eth,debug',
+        '--verbosity', '4', '--gcmode', 'archive', '--dev.gaslimit', '30000000'
+    ])
+
+    forge = multiprocessing.Process(target=deploy_contracts(paths))
+    forge.start()
+    forge.join()
+
+    res = debug_dumpBlock('127.0.0.1:8545')
+    response = json.loads(res)
+    allocs = response['result']
+
+    write_json(paths.allocs_path, allocs)
+    geth.terminate()
 
-    log.info('Bringing up the services.')
-    run_command(['docker-compose', 'up', '-d', 'op-proposer', 'op-batcher'], cwd=paths.ops_bedrock_dir, env={
-        'PWD': paths.ops_bedrock_dir,
-        'L2OO_ADDRESS': '0x6900000000000000000000000000000000000000'
-    })
 
 # Bring up the devnet where the contracts are deployed to L1
 def devnet_deploy(paths):
     if os.path.exists(paths.genesis_l1_path):
         log.info('L1 genesis already generated.')
     else:
         log.info('Generating L1 genesis.')
-        write_json(paths.genesis_l1_path, GENESIS_TMPL)
+        if os.path.exists(paths.allocs_path) == False:
+            devnet_l1_genesis(paths)
```

Comment:
what is the difference between `allocs_path` and `genesis_l1_path`? Why do we appear to be generating `allocs_path` with the function `devnet_l1_genesis`?

Reply:
The allocs are the result of `debug_dumpBlock` so its the state dump, its called allocs in the genesis struct